lib/Service/RoomService.php

```diff
@@ -547,18 +547,45 @@ public function setMessageExpiration(Room $room, Participant $participant, int $
 			->set('message_expiration', $update->createNamedParameter($seconds, IQueryBuilder::PARAM_INT))
 			->where($update->expr()->eq('id', $update->createNamedParameter($room->getId(), IQueryBuilder::PARAM_INT)));
 		$update->executeStatement();
+
 		$room->setMessageExpiration($seconds);
+
+		$this->toggleJobExpireMesssageIfNecessary($seconds);
+		$this->addMessageExpirationSystemMessage($room, $participant, $seconds);
+
+		$this->dispatcher->dispatch(Room::EVENT_AFTER_SET_MESSAGE_EXPIRATION, $event);
+	}
+
+	private function toggleJobExpireMesssageIfNecessary(int $seconds): void {
```

Comment:
Remove this function. We will just hard code the job and always run it.
We also have the job to expire internal signaling messages even when the HPB is used.
For that you can simply define the job in info.xml and bump the version:

https://github.com/nextcloud/spreed/blob/26808039bd1f967cf6ce81b9be3c822dfd345d9b/appinfo/info.xml#L59